indigo-frontend/src/app/pages/experiment/sample-search/sample-search.component.ts

```diff
@@ -0,0 +1,327 @@
+    FormDialogComponent,
+    InputComponent,
+    MatRadioGroup,
+    MatRadioButton,
+    MatExpansionPanel,
+    MatExpansionPanelHeader,
+    MatExpansionPanelTitle,
+    MatExpansionPanelDescription,
+    TextSearchComponent,
+    NumericSearchComponent,
+    DropdownValueComponent,
+    MatChip,
+    MatProgressSpinner,
+    ExpandableTableComponent,
+    ColumnDefDirective,
+    ApiImageComponent,
+    InfiniteLoaderComponent,
+    MatTooltip,
+    ToggleComponent,
+  ],
+  templateUrl: './sample-search.component.html',
+})
+export class SampleSearchComponent implements OnInit {
+  data: SampleSearchDialogData = inject(MAT_DIALOG_DATA);
+
+  loader: InfiniteSearchLoader<FindSamplesRequest, Sample>;
+
+  @ViewChild('advancedSearchPanel') advancedSearchPanel: MatExpansionPanel;
+
+  service = inject(ApiService);
+  destroyRef = inject(DestroyRef);
+  dialog = inject(MatDialog);
+  experimentModelService = inject(ExperimentModelService);
+
+  title = 'Add Material';
+
+  form = new FormGroup({
+    quickSearch: new FormControl<string | null>(null),
+    structureSearchType: new FormControl<StructuralSearchType>(
+      StructuralSearchType.EXACT,
+    ),
+    strCode: new FormControl<TextSearch | null>(null),
+    nbkBatchNumber: new FormControl<TextSearch | null>(null),
+    molecularFormula: new FormControl<TextSearch | null>(null),
+    molWeight: new FormControl<NumericSearch | null>(null),
+    chemicalName: new FormControl<TextSearch | null>(null),
+    compoundState: new FormControl<DictionaryItemRef | null>(null),
+    batchComment: new FormControl<TextSearch | null>(null),
+    healthHazards: new FormControl<DictionaryItemRef | null>(null),
+    casNumber: new FormControl<TextSearch | null>(null),
+    marked: new FormControl<boolean>(false),
+  });
+  lastMarked = this.form.value.marked;
+  structureMolFile: string | null = null;
+  structureImage: string | null = null;
+
+  advancedSearchSummary: string[][] | null = null;
+  compoundStateOptions: DictionaryItemRef[];
+  healthHazardsOptions: DictionaryItemRef[];
+
+  ngOnInit(): void {
+    this.loader = new InfiniteSearchLoader<FindSamplesRequest, Sample>(
+      (searchParams, pageNo) =>
+        this.service.request(
+          'post',
+          `samples/search?pageNo=${pageNo}&pageSize=20`,
+          searchParams,
+        ),
+    );
+    this.service
+      .request<
+        DictionaryItemRef[]
+      >('get', `dictionaries/${BuiltInDictionary.COMPONENT_STATE}`)
+      .subscribe((list) => {
+        this.compoundStateOptions = list;
+      });
+    this.service
+      .request<
+        DictionaryItemRef[]
+      >('get', `dictionaries/${BuiltInDictionary.HEALTH_HAZARD}`)
+      .subscribe((list) => {
+        this.healthHazardsOptions = list;
+      });
+    this.form.valueChanges.pipe(takeUntilDestroyed(this.destroyRef)).subscribe({
+      next: (value) => {
+        console.log('Form changed: ', value);
+        if (value.marked !== this.lastMarked) {
```

Comment:
Could you please check if we can use rxjs operator `distinctUntilChanged` before `takeUntilDestroyed` instead of this.